src/util/pup.h

```diff
@@ -126,6 +127,37 @@ typedef enum {
   dataType_last //<- for setting table lengths, etc.
 } dataType;
 
+template <class T>
+dataType getXlateDataType(T *a) {
+  if(std::is_same<T, signed char>::value) return Tchar;
```

Comment:
Maybe better to use a switch/case here

Reply:
I am wondering if I can use `is_same` expression in a switch case because the expression evaluates to boolean and not distinct values that I can compare in the switch statement.